cloud/amazon/cloudformation_facts.py

```diff
@@ -0,0 +1,272 @@
+    returned: only if all_facts or stack_resourses is true
+    type: dict
+'''
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+from functools import partial
+import json
+
+class CloudFormationServiceManager:
+    """Handles CloudFormation Services"""
+
+    def __init__(self, module):
+        self.module = module
+
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+            self.client = boto3.client('cloudformation', region_name=region, **aws_connect_kwargs)
+        except botocore.exceptions.NoRegionError:
+            self.module.fail_json(msg="Region must be specified as a parameter, in AWS_DEFAULT_REGION environment variable or in boto configuration file")
+        except boto.exception.NoAuthHandlerFound as e:
+            self.module.fail_json(msg="Can't authorize connection - "+str(e))
+
+    def describe_stack(self, stack_name):
+        try:
+            func = partial(self.client.describe_stacks,StackName=stack_name)
+            response = self.paginated_response(func, 'Stacks')
+            if response:
+                return response[0]
+            self.module.fail_json(msg="Error describing stack - an empty response was returned")
+        except Exception as e:
+            self.module.fail_json(msg="Error describing stack - " + str(e))
+    
+    def list_stack_resources(self, stack_name):
+        try:
+            func = partial(self.client.list_stack_resources,StackName=stack_name)
+            return self.paginated_response(func, 'StackResourceSummaries')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing stack resources - " + str(e))
+
+    def describe_stack_events(self, stack_name):
+        try:
+            func = partial(self.client.describe_stack_events,StackName=stack_name)
+            return self.paginated_response(func, 'StackEvents')
+        except Exception as e:
+            self.module.fail_json(msg="Error describing stack events - " + str(e))
+
+    def get_stack_policy(self, stack_name):
+        try:
+            response = self.client.get_stack_policy(StackName=stack_name)
+            stack_policy = response.get('StackPolicyBody')
+            if stack_policy:
+                return json.loads(stack_policy)
+            return dict()
+        except Exception as e:
+            self.module.fail_json(msg="Error getting stack policy - " + str(e))
+
+    def get_template(self, stack_name):
+        try:
+            response = self.client.get_template(StackName=stack_name)
+            return response.get('TemplateBody')
+        except Exception as e:
+            self.module.fail_json(msg="Error getting stack template - " + str(e))
```

Comment:
For all these fails, can you include the traceback like so:

```
try:
    # some stuff
except Exception as e:
    self.module.fail_json(msg="Error getting stack template - " + str(e), exception=traceback.format_exc(e))
```
And `import traceback` at the top of the file?